aiida_quantumespresso/workflows/pw/bands.py

```diff
@@ -168,8 +170,13 @@ def run_scf(self):
         inputs.metadata.call_link_label = 'scf'
         inputs.pw.structure = self.ctx.current_structure
         inputs.pw.parameters = inputs.pw.parameters.get_dict()
-        inputs.pw.parameters.setdefault('CONTROL', {})
-        inputs.pw.parameters['CONTROL']['calculation'] = 'scf'
+        inputs.pw.parameters.setdefault('CONTROL', {})['calculation'] = 'scf'
+
+        # Make sure to carry the number of bands from the relax workchain if it was run and it wasn't explicitly defined
+        # in the inputs. One of the base workchains in the relax workchain may have changed the number automatically in
+        #  the sanity checks on band occupations.
+        if self.ctx.current_number_of_bands:
```

Comment:
Maybe this check should be `if self.ctx.current_number_of_bands is not None:`? Right now all values that convert to `False` are ignored.

Then again, a `nbnd` of zero isn't really valid, so it probably doesn't matter 😄

Reply:
That was my thinking exactly